File: PhotoOverlay/src/cc/android/photooverlay/Splash.java
package cc.android.photooverlay;

import cc.android.photooverlay.BillingTask.Op;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;

public class Splash extends BaseActivity {

	@Override
	public void onCreate(Bundle bundle) {
		super.onCreate(bundle);
		View v = View.inflate(this, R.layout.splash, null);
		setContentView(v);
		v.postDelayed(new Runnable() {
			public void run() {
				startActivity(new Intent(getActivity(), FormsList.class));
				finish();
			}
		}, 4000);
		new BillingTask(Op.REFRESH_PURCHASED, getActivity()).execute();
	}
	
}

File: Thomas/src/main/java/cc/android/thomas/Splash.kt
package cc.android.thomas

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.os.Handler
import cc.android.thomas.databinding.SplashBinding

class Splash : Activity(), Runnable {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
	    with (SplashBinding.inflate(layoutInflater)) {
	    	setContentView(root)
		    screen.setOnClickListener { run() }
	    }
        handler.postDelayed(this, if (BuildConfig.DEBUG) 2000 else 5000.toLong())
    }

    var handler = Handler()

    override fun onPause() {
        super.onPause()
        handler.removeCallbacks(this)
    }

    override fun run() {
        val intent = Intent(this, MainActivity::class.java)
        startActivity(intent)
        finish()
    }
}